In [3]:
# 🏠 Real Estate ROI Calculator

def calculate_roi(price_lakhs, rent, expenses):
    price = price_lakhs * 100000  # convert lakhs to rupees
    annual_rent = rent * 12
    net_income = annual_rent - expenses
    roi = (net_income / price) * 100
    return round(roi, 2)

print("🏠 Real Estate ROI Calculator\n")

# Inputs
price = float(input("Enter Property Price (in Lakhs): "))
rent = float(input("Enter Monthly Rent: "))
expenses = float(input("Enter Annual Expenses: "))

# Calculation
roi = calculate_roi(price, rent, expenses)

# Decision Logic
if roi > 8:
    decision = "BUY ✅"
elif roi > 4:
    decision = "HOLD ⚖️"
else:
    decision = "AVOID ❌"

# Output
print("\n📊 RESULT")
print("ROI:", roi, "%")
print("Decision:", decision)

🏠 Real Estate ROI Calculator

Enter Property Price (in Lakhs): 50
Enter Monthly Rent: 15000
Enter Annual Expenses: 20000

📊 RESULT
ROI: 3.2 %
Decision: AVOID ❌


Perfect 👍 — here are 3 clean, professional test cases you can use in your GitHub README or demo.

⸻

🧪 TEST CASE 1 — Good Investment (BUY)

Input:

Price: 30 Lakhs
Rent: 25000
Expenses: 20000

Calculation:

* Annual Rent = 25,000 × 12 = 3,00,000
* Net Income = 3,00,000 − 20,000 = 2,80,000

Output:

ROI: 9.33 %
Decision: BUY ✅

⸻

🧪 TEST CASE 2 — Medium Investment (HOLD)

Input:

Price: 40 Lakhs
Rent: 20000
Expenses: 20000

Calculation:

* Annual Rent = 2,40,000
* Net Income = 2,20,000

Output:

ROI: 5.5 %
Decision: HOLD ⚖️

⸻

🧪 TEST CASE 3 — Bad Investment (AVOID)

Input:

Price: 50 Lakhs
Rent: 15000
Expenses: 20000

Calculation:

* Annual Rent = 1,80,000
* Net Income = 1,60,000

Output:

ROI: 3.2 %
Decision: AVOID ❌

⸻

💡 Pro Tip (Add This in README 🔥)

Add section like:

## 🧪 Test Cases
| Price (Lakhs) | Rent | Expenses | ROI | Decision |
|--------------|------|----------|-----|----------|
| 30 | 25000 | 20000 | 9.33% | BUY |
| 40 | 20000 | 20000 | 5.5% | HOLD |
| 50 | 15000 | 20000 | 3.2% | AVOID |

👉 This makes your project look professional to recruiters

⸻

✅ Quick Summary

* 3 test cases:
    * High ROI → BUY
    * Medium → HOLD
    * Low → AVOID
* Add to README → strong portfolio 💼

⸻

🤔 Suggested Follow-Up Questions

* Want EMI calculator test cases next?
* Want auto-test script (run all cases in code)?
* Want convert this into table output in Colab?
* Want next project for GitHub (build strong portfolio)?

⸻

Now your project is:
👉 Tested + Presentable + Job-ready 🚀

Great question — this is where your project becomes real-world useful 👍

Right now you used:

* 8% → BUY
* 4–8% → HOLD
* < 4% → AVOID

👉 But let’s understand why these limits exist and how to make them professional.

⸻

🧠 Real Estate ROI Benchmarks (Reality)

🥇 High ROI (GOOD)

* 8% – 12%+
* Strong rental income
* Good investment

👉 Decision: BUY ✅

⸻

⚖️ Medium ROI (OK)

* 4% – 8%
* Average returns
* Depends on future appreciation

👉 Decision: HOLD ⚖️

⸻

❌ Low ROI (BAD)

* Below 4%
* Poor rental return
* Not worth risk

👉 Decision: AVOID ❌

⸻

💡 Why These Limits?

Because:

Compare with alternatives:

Investment	Return
Fixed Deposit	~6–7%
Real Estate (rent)	~2–10%
Stocks	~10–15%

👉 So:

* If ROI < FD → ❌ bad
* If ROI ≈ FD → ⚖️ average
* If ROI > FD → ✅ good

⸻

🔥 Make Your App SMART (Upgrade)

Instead of hardcoding:

if roi > 8:

👉 Make it configurable:

BUY_THRESHOLD = 8
HOLD_THRESHOLD = 4
if roi > BUY_THRESHOLD:
    decision = "BUY ✅"
elif roi > HOLD_THRESHOLD:
    decision = "HOLD ⚖️"
else:
    decision = "AVOID ❌"

⸻

🚀 Advanced Version (Pro Level)

Let user decide:

buy_threshold = float(input("Enter BUY threshold (%): "))
hold_threshold = float(input("Enter HOLD threshold (%): "))

👉 Now your tool becomes:

* Flexible
* Customizable
* More valuable 💰

⸻

⚠️ Important Real-World Factors (Beyond ROI)

Tell this in your README (very important for jobs 💼):

ROI alone is not enough:

* Location growth
* Property appreciation
* Vacancy risk
* Maintenance cost

👉 This shows real-world understanding

⸻

🎯 Best Default Settings (Use This)

BUY_THRESHOLD = 8
HOLD_THRESHOLD = 4

👉 This is:

* Simple
* Industry-aligned
* Easy to explain

⸻

✅ Quick Summary

* ROI thresholds are based on market comparison
* < 4% → bad
* 4–8% → average
* 8% → good
* Make thresholds configurable for better product

⸻

🤔 Suggested Follow-Up Questions

* Want add EMI + ROI combined decision (very powerful)?
* Want show ROI vs FD comparison in output?
* Want add property appreciation logic?
* Want make this tool look like premium SaaS?

⸻

This step is important 👇
👉 You moved from “coding logic” → to business logic 🚀